data/settings/src/main/java/com/google/jetpackcamera/settings/model/CameraAppSettings.kt

```diff
@@ -30,8 +30,16 @@ data class CameraAppSettings(
     val dynamicRange: DynamicRange = DynamicRange.SDR,
     val defaultHdrDynamicRange: DynamicRange = DynamicRange.HLG10,
     val zoomScale: Float = 1f,
-    val targetFrameRate: Int = TARGET_FPS_AUTO
-)
+    val targetFrameRate: Int = TARGET_FPS_AUTO,
+    val imageFormat: ImageOutputFormat = ImageOutputFormat.JPEG
+) {
+    fun isCombinationSupported(): Boolean {
+        // Makes Ultra HDR and Single Stream Mode features mutually exclusive, since single-stream
+        // mode uses CameraEffect, which does not support HDR-related functions.
+        return imageFormat != ImageOutputFormat.JPEG_ULTRA_HDR ||
+            captureMode != CaptureMode.SINGLE_STREAM
+    }
+}
```

Comment:
Thank you, updated with the following changes:
1. exposed the mutually exclusive information by using `supportedImageFormatsMap` (`Map<CaptureMode, Set<ImageOutputFormat>>`) instead of `supportedImageFormats` in `CameraConstraints`.
2. change mode to `SINGLE_STREAM` will reset `imageFormat` to `JPEG` instead of failing.
 
Since HDR video may still be supported in single stream mode, the HDR button may not be deactivated, the UI control approach is slightly different, disabling the toggle button instead of the quick settings button:
1. For capture mode switching: check image format and reset it if needed.
2. For image format switching: will not affect capture mode. Since the toggle button enable/disable is based on the `supportedImageFormatsMap`, the image format cannot be set to Ultra HDR in single-stream mode (`supportedImageFormats` contains only `JPEG`).
 
Screenshot of `Single-stream Mode + HDR`:
<img src="https://github.com/google/jetpack-camera-app/assets/37896299/c6e01ad9-ae0e-4b23-a559-7279c809096a" width="300" />  

P.S.: Since the testing device (S24) crashes when enabling HDR video in single stream mode ([issue#173](https://github.com/google/jetpack-camera-app/issues/173)), the check is done with VideoCapture not actually bound.

PTAL.